gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/transformation/Transformation.java

```diff
@@ -59,66 +68,58 @@
-    TransformationFunction<Edge> edgeTransFunc) {
+  public Transformation(
+    TransformationFunction<G> graphHeadTransFunc,
+    TransformationFunction<V> vertexTransFunc,
+    TransformationFunction<E> edgeTransFunc) {
 
-    if (graphHeadTransFunc == null &&
-      vertexTransFunc == null &&
-      edgeTransFunc == null) {
-      throw new IllegalArgumentException(
-        "Provide at least one transformation function.");
+    if (graphHeadTransFunc == null && vertexTransFunc == null && edgeTransFunc == null) {
+      throw new IllegalArgumentException("Provide at least one transformation function.");
     }
     this.graphHeadTransFunc = graphHeadTransFunc;
     this.vertexTransFunc    = vertexTransFunc;
     this.edgeTransFunc      = edgeTransFunc;
   }
 
   @Override
-  public LogicalGraph execute(LogicalGraph graph) {
+  public LG execute(LG graph) {
     return executeInternal(
       graph.getGraphHead(),
       graph.getVertices(),
       graph.getEdges(),
-      graph.getConfig());
+      graph.getFactory());
   }
 
   /**
    * Applies the transformation functions on the given datasets.
    *
-   * @param graphHeads  graph heads
-   * @param vertices    vertices
-   * @param edges       edges
-   * @param config      gradoop flink config
+   * @param graphHeads graph heads
+   * @param vertices vertices
+   * @param edges edges
+   * @param factory the factory that is responsible for creating an instance of the logical graph
    * @return transformed logical graph
    */
   @SuppressWarnings("unchecked")
-  protected LogicalGraph executeInternal(DataSet<GraphHead> graphHeads,
-    DataSet<Vertex> vertices, DataSet<Edge> edges, GradoopFlinkConfig config) {
+  protected LG executeInternal(DataSet<G> graphHeads, DataSet<V> vertices, DataSet<E> edges,
+    BaseGraphFactory<G, V, E, LG> factory) {
 
-    DataSet<GraphHead> transformedGraphHeads = graphHeadTransFunc != null ?
-      graphHeads.map(new TransformGraphHead(
-        graphHeadTransFunc, config.getGraphHeadFactory()))
-        .returns(TypeExtractor.createTypeInfo(
-          config.getGraphHeadFactory().getType())) : graphHeads;
+    DataSet<G> transformedGraphHeads = graphHeadTransFunc != null ?
+      graphHeads
+        .map(new TransformGraphHead(graphHeadTransFunc, factory.getGraphHeadFactory()))
+        .returns(TypeExtractor.createTypeInfo(factory.getGraphHeadFactory().getType())) :
+      graphHeads;
 
-    DataSet<Vertex> transformedVertices = vertexTransFunc != null ?
-      vertices.map(new TransformVertex(
-        vertexTransFunc, config.getVertexFactory()))
-        .returns(TypeExtractor.createTypeInfo(
-          config.getVertexFactory().getType())) : vertices;
+    DataSet<V> transformedVertices = vertexTransFunc != null ?
+      vertices
+        .map(new TransformVertex(vertexTransFunc, factory.getVertexFactory()))
+        .returns(TypeExtractor.createTypeInfo(factory.getVertexFactory().getType())) : vertices;
```

Comment:
Move vertices to next line to make else statement more visible (and consistent with graphHeads in previous one).